packages/react-codesandboxer/src/CodeSandboxer.js

```diff
@@ -188,6 +188,25 @@ export default class CodeSandboxDeployer extends Component<Props, State> {
     }
   };
 
+  componentDidUpdate(prevProps: Props) {
+    if (
+      this.props.examplePath !== prevProps.examplePath ||
+      this.props.name !== prevProps.name ||
+      this.props.gitInfo !== prevProps.gitInfo ||
+      this.props.example !== prevProps.example ||
+      this.props.pkgJSON !== prevProps.pkgJSON ||
+      this.props.importReplacements !== prevProps.importReplacements ||
+      this.props.dependencies !== prevProps.dependencies ||
+      this.props.providedFiles !== prevProps.providedFiles ||
+      this.props.extensions !== prevProps.extensions ||
+      this.props.template !== prevProps.template
```

Comment:
Two concerns with this approach:
- First is that using a comparator operator for objects scares me. Comparing `this.props.pkgJSON !== prevProps.pkgJSON` for example should always be `true`, because the two objects will have different references. Something like lodashe's [isEqual](https://lodash.com/docs/4.17.11#isEqual) function can solve this.
- Second I'm always a bit wary of `componentDidUpdate`. I'm tempted to hash these values and save that to state, so the logic can just occur at the start of deployToCSB as just

```js
if (deployPromise && this.state.hash === getNewHash(this.props) {
      deployPromise.then(this.deploy);
} else {
      this.loadFiles().then(this.deploy);
}
```

then loadFiles sets the hash state every time it's called.

Biggest problem is I can't find a hashing library that doesn't add a bunch of page weight, so maybe we can't do this.

Reply:
Good point, currently changing to use lodash.isEqual. Just discovered though that because props.example and props.pkgJSON are allowed to be promises, lodash.equal can't equate them in the way I would like them to.

Perhaps hashing in deployToCSB could fix this too.